authorisation-adjustment-example/src/main/java/com/adyen/checkout/api/ApiController.java

```diff
@@ -0,0 +1,189 @@
+        this.paymentsApi = new PaymentsApi(client);
+    }
+
+    /**
+     * {@code POST  /getPaymentMethods} : Get valid payment methods.
+     *
+     * @return the {@link ResponseEntity} with status {@code 200 (Ok)} and with body the paymentMethods response.
+     * @throws IOException  from Adyen API.
+     * @throws ApiException from Adyen API.
+     */
+    @PostMapping("/getPaymentMethods")
+    public ResponseEntity<PaymentMethodsResponse> paymentMethods() throws IOException, ApiException {
+        var paymentMethodsRequest = new PaymentMethodsRequest();
+        paymentMethodsRequest.setMerchantAccount(this.applicationProperty.getMerchantAccount());
+        paymentMethodsRequest.setChannel(PaymentMethodsRequest.ChannelEnum.WEB);
+
+        log.info("REST request to get Adyen payment methods {}", paymentMethodsRequest);
+        var response = paymentsApi.paymentMethods(paymentMethodsRequest);
+        return ResponseEntity.ok()
+            .body(response);
+    }
+
+    /**
+     * {@code POST  /initiatePayment} : Make a payment.
+     *
+     * @return the {@link ResponseEntity} with status {@code 200 (Ok)} and with body the paymentMethods response.
+     * @throws IOException  from Adyen API.
+     * @throws ApiException from Adyen API.
+     */
+    @PostMapping("/pre-authorisation")
+    public ResponseEntity<PaymentResponse> payments(@RequestHeader String host, @RequestBody PaymentRequest body, HttpServletRequest request) throws IOException, ApiException {
+        var paymentRequest = new PaymentRequest();
+
+        var orderRef = UUID.randomUUID().toString();
+        var amount = new Amount()
+            .currency("EUR")
+            .value(24999L); // value is 249.99€ in minor units
+
+        paymentRequest.setMerchantAccount(this.applicationProperty.getMerchantAccount()); // required
+        paymentRequest.setChannel(PaymentRequest.ChannelEnum.WEB);
+        paymentRequest.setReference(orderRef); // required
+        paymentRequest.setReturnUrl(request.getScheme() + "://" + host + "/api/handleShopperRedirect?orderRef=" + orderRef);
+
+        paymentRequest.setAmount(amount);
+        var authenticationData = new AuthenticationData();
+        authenticationData.setAttemptAuthentication(AuthenticationData.AttemptAuthenticationEnum.ALWAYS);
+        paymentRequest.setAuthenticationData(authenticationData);
+
+        paymentRequest.setAdditionalData(new HashMap<>() { {
+            put ("authorisationType", "PreAuth");
+        }});
+
+        // required for 3ds2 redirect flow
+        paymentRequest.setOrigin(request.getScheme() + "://" + host);
+        // required for 3ds2
+        paymentRequest.setBrowserInfo(body.getBrowserInfo());
+        // required by some issuers for 3ds2
+        paymentRequest.setShopperIP(request.getRemoteAddr());
+        paymentRequest.setPaymentMethod(body.getPaymentMethod());
+
+        // we strongly recommend that you the billingAddress in your request
```

Comment:
We should  remove the commented code (and comment) or elaborate with providing a good example